### PR TITLE
chore: Make ErrorPage generic with basic defaults

### LINKED ISSUE #1733: Create a generic error handler in core and shared more code between error components
There is code in the Error components that can be cleaned up/shared/used in Disco Pane (and eventually the Admin too) but isn't yet. This was mentioned in #1685 and I have a PR incoming; just logging here so there's an issue.

### PR DESCRIPTION
**Pull request by tofumatt**
Fixes disco pane errors and improves error handling for apps without their own error components.

Moves content (like suggested pages) into their own component.

Also adds a 404 route for the Disco Pane and similar dev routes for `/404` and `/500` for testing in dev mode as in the `amo` app.

Fixes #1733.

Sorry you have to review more errors, @kumar303 😉

-----

## New Disco pane error pages
screen shot 2017-02-07 at 14 18 45
screen shot 2017-02-07 at 14 18 39

### CODE REVIEW

File: src/core/components/ErrorPage/GenericError/index.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    const { i18n, status } = this.props;
+
+    return (
+      <NestedStatus code={status}>
+        <div className="ErrorPage GenericError">
+          <h1>{i18n.gettext('Server Error')}</h1>
+
+          <p>
+            {i18n.gettext(oneLine`
+              Sorry, but there was an error we couldn't complete your request.
```
Comment (kumar303) [inline src/core/components/ErrorPage/GenericError/index.js]: typo: Sorry, but there was an error and we couldn't...

File: src/core/components/ErrorPage/NotFound/index.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  render() {
+    const { i18n, status } = this.props;
+
+    return (
+      <NestedStatus code={status}>
+        <div className="ErrorPage NotFound">
+          <h1>{i18n.gettext('Page not found')}</h1>
+          <p>
+            {i18n.gettext(oneLine`
+              Sorry, but we can't find anything at the address you entered.`)}
```
Comment (kumar303) [inline src/core/components/ErrorPage/NotFound/index.js]: I think 'URL' would be better than 'address'. It's more concrete and actionable.

File: src/core/components/ErrorPage/index.js [hunk 3]
```diff
@@ -2,21 +2,23 @@ import React, { PropTypes } from 'react';
 import { connect } from 'react-redux';
 import { compose } from 'redux';
 
-import { getErrorComponent } from 'amo/utils';
+import { getErrorComponent as getErrorComponentCore } from 'core/utils';
```
Comment (kumar303) [inline src/core/components/ErrorPage/index.js]: This looks a bit more straightforward to me but I guess either one is slightly awkward:
````js
import { getErrorComponent as defaultErrorComponentGetter } from 'core/utils';
````
Meh, words.

File: src/disco/containers/App.js [hunk 4]
```diff
@@ -29,7 +35,9 @@ export class AppBase extends React.Component {
             { name: 'robots', content: 'noindex' },
           ]}
         />
-        {children}
+        <ErrorPage>
```
Comment (kumar303) [inline src/disco/containers/App.js]: ooh, I'm scared. I trust you though.

File: src/core/components/ErrorPage/index.js (hunk 3 above)
Comment (tofumatt) [inline src/core/components/ErrorPage/index.js]: `getErrorComponentDefault` would be fine (and in line with what we do elsewhere), I don't know why I picked this.

File: src/disco/containers/App.js (hunk 4 above)
Comment (tofumatt) [inline src/disco/containers/App.js]: trust fall refreshments

File: src/core/components/ErrorPage/index.js (hunk 3 above)
Comment (kumar303) [inline src/core/components/ErrorPage/index.js]: that works! And it's shorter

File: src/core/components/ErrorPage/NotFound/index.js (hunk 2 above)
Comment (tofumatt) [inline src/core/components/ErrorPage/NotFound/index.js]: Agreed. I thought address was less nerdy but URL is more concrete. Thanks 👍

File: src/core/components/ErrorPage/GenericError/index.js (hunk 1 above)
Comment (tofumatt) [inline src/core/components/ErrorPage/GenericError/index.js]: Merci !

File: src/disco/containers/App.js (hunk 4 above)
Comment (kumar303) [inline src/disco/containers/App.js]: omg, stealing this gif. So good.

**Pull request merged**